src/main/sampling/Sample.cpp

```diff
@@ -182,6 +182,122 @@ namespace lsp
             return true;
         }
 
+        bool Sample::stretch(size_t channels, int stretch_samples, size_t chunk_size)
+        {
+            if (channels <= 0)
+                return false;
+
+            size_t max_length = nMaxLength;
+            size_t length = nLength;
+
+            // Allocate new data
+            max_length      = align_size(max_length, DEFAULT_ALIGN);    // Make multiple of 4
+            size_t to_copy      = (nMaxLength > max_length) ? max_length : nMaxLength;
+
+            size_t nlen = lsp_max((int)to_copy-stretch_samples, 0);
+            if (nlen <= 0) 
+              return false;
+
+            max_length = nlen;
+            length = nlen;
+
+            float *buf      = reinterpret_cast<float *>(::malloc(max_length * channels * sizeof(float)));
+            if (buf == NULL)
+                return false;
+
+            // Copy previously allocated data
+            if (vBuffer != NULL)
+            {
+                float *dptr         = buf;
+                const float *sptr   = vBuffer;
+
+                // Copy channels
+                for (size_t ch=0; ch < channels; ++ch)
+                {
+                    if (ch < nChannels)
+                    {
+                        size_t n_chunks = to_copy/chunk_size;
+                        size_t n_chunks_dst = nlen/chunk_size;
+
+                        for (size_t i = 0; i < nlen; i++) {
+                            dptr[i] = 0;
+                        }
+
+                        if (stretch_samples > 0) {
+                            /**
+                             *  STRETCH DOWN
+                             */
+                            float stretch_ratio = (float)n_chunks/n_chunks_dst;
+                          
+                            for (size_t c = 0; c < n_chunks_dst; c++) {
+                                for (size_t i = 0; i < chunk_size; i++) {
+                                    if (c == 0) {
```

Comment:
This is not good for optimizations. The check for the number of chunk should be moved outside of the inner loop. Otherwise these checks will blow up the branch predictor.